docs/standard/commandline/model-binding.md

```diff
@@ -183,30 +191,30 @@ Besides the file system types and `Uri`, the following types are supported:
 
 ## Inject System.CommandLine types
 
-`System.CommandLine` allows you to use some types in handlers by adding parameters for them to the handler signature. The available types include:
+`System.CommandLine` allows you to use some types in handlers by providing a `SetHandler` overload that gives you access to the <xref:System.CommandLine.Invocation.InvocationContext>. The available types include:
 
+* <xref:System.CommandLine.Invocation.InvocationContext>
 * <xref:System.Threading.CancellationToken>
 * <xref:System.CommandLine.IConsole>
-* <xref:System.CommandLine.Invocation.InvocationContext>
 * <xref:System.CommandLine.Parsing.ParseResult>
 
 Other types can be injected by using custom binders. For more information, see [Dependency injection](dependency-injection.md).
```

Comment:
I wouldn't call this injection anymore, since (IIRC correct me if I'm wrong here @jonsequitur), there's no DI of the System.CommandLine types anymore. If you want the cancellationToken, you need to get the InvocationContext and access the property on it. If you want an IConsole, you need to get the InvocationContext and access the property on it. And so on.

Reply:
I didn't actually remove those yet but it's probably a good idea. `SetHandler` won't inject them anymore in any case.